pytest_html/plugin.py

```diff
@@ -215,23 +218,51 @@ def pytest_sessionfinish(self, session):
             html.title('Test Report'),
             html.style(raw(style_css)))
 
+
+        class ResultByType:
```

Comment:
This is fine. We could also move it to it's own file. In this case is the type the outcome? I'm a bit confused by the name of the class. This object represents each potential outcome, perhaps `Outcome` would be a better name?